src/app/body/wishlist/wishlist.resolver.ts

```diff
@@ -0,0 +1,14 @@
+import { Injectable } from '@angular/core';
+import { Resolve } from '@angular/router';
+import { Product } from '../../models/product.model';
+import { WishlistService } from '../../services/wishlist.service';
+
+
+@Injectable()
+export class WishlistResolver implements Resolve<Promise<Product[]>> {
+  constructor(private wishlistService: WishlistService) {}
+
+  resolve(): Promise<Product[]> {
+    return this.wishlistService.getWishlist();
```

Comment:
Use this data that you pass through resolver in your component (like as you used `resolver` with products) instead of using it from services/subscriptions in your `WishlistComponent`

Reply:
Can I subscribe to this resolver data?